### PR TITLE
Stream print & print bytes

### PR DESCRIPTION
**Pull request by Mackey22**
* This handles data formats that the cli previously didn't decode (specifically gzip). 
* Also handles the case where you call get-files on an enormous file. Streams the content instead of loading it immediately. 
* Changes the `make test` target to include install, which pip installs the package and puts hca into the namespace. I couldn't think of another way to check that the output printed bytes - open to other suggestions if you think this is a bad solution.

### CODE REVIEW

File: hca/__init__.py [hunk 1]
```diff
@@ -6,13 +6,21 @@
...
     response = cli.make_request(sys.argv[1:])
     if isinstance(response, requests.Response):
-        print(response.content.decode())
+
+        for chunk in response.iter_content(chunk_size=Constants.CHUNK_SIZE, decode_unicode=True):
+            if chunk:  # filter out keep-alive new chunks
+                try:
+                    print(chunk.decode())
+                except UnicodeDecodeError:
+                    print(chunk)
```
Comment (kislyuk) [inline hca/__init__.py]: Nope, don't decode!

Comment (Mackey22) [inline hca/__init__.py]: Resolved

**Comment by ttung** [comment]
This looks good.  Does travis have a mechanism by which we can do CI on windows?

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ lint:
 hca/api_spec.json:
 	curl https://hca-dss.czi.technology/v1/swagger.json > hca/api_spec.json
 
-test: lint bindings
+test: lint install
```
Comment (ttung) [inline Makefile]: why this change?  test doesn't regenerate bindings?

File: hca/__init__.py [hunk 3]
```diff
@@ -3,16 +3,36 @@
...
     else:
         print(json.dumps(response))
+
+
+def _write_binary_to_stdout(content):
+    if six.PY3:
+        sys.stdout.buffer.write(content)
+    else:
+        # Windows includes carriage returns
+        if sys.platform == 'win32':
```
Comment (ttung) [inline hca/__init__.py]: wouldn't it make sense to do this once before your for loop?

File: test/test.py [hunk 4]
```diff
@@ -22,6 +22,8 @@
 from hca import constants  # noqa
 from test.config import override_oauth_config  # noqa
 
+dirpath = os.path.dirname(os.path.realpath(__file__))
```
Comment (ttung) [inline test/test.py]: why not move this just to `test_get_files_cli`?

File: hca/__init__.py (hunk 3 above)
Comment (Mackey22) [inline hca/__init__.py]: True. Will change

File: test/test.py (hunk 4 above)
Comment (Mackey22) [inline test/test.py]: Was going to refactor for all tests using `dirpath` to just use this but didn't want to clutter the pr. Will just move to `test_get_files_cli` for now.

File: Makefile (hunk 2 above)
Comment (Mackey22) [inline Makefile]: Tests now rely on calling `hca get-files` cmd in a subprocess, so that command has to be installed. Thoughts on this pattern overall?

Comment (ttung) [inline Makefile]: Oh I see.  Install relies on bindings.  Carry one.

**Comment by ttung** [comment]
This diff is much larger than it was maybe an hour ago when I saw it last, FYI.

**Comment by Mackey22** [comment]
Yeah, the branch underneath was rebased - need to rebase this one.

**Comment by Mackey22** [comment]
Can't easily do CI on Windows: https://github.com/travis-ci/travis-ci/issues/216

Ready for review.

File: hca/__init__.py [hunk 5]
```diff
@@ -3,16 +3,36 @@
...
     if isinstance(response, requests.Response):
-        print(response.content.decode())
+        for chunk in response.iter_content(chunk_size=Constants.CHUNK_SIZE, decode_unicode=True):
+            if chunk:  # filter out keep-alive new chunks
+                _write_binary_to_stdout(chunk)
     else:
         print(json.dumps(response))
+
+
+def _write_binary_to_stdout(content):
```
Comment (ttung) [inline hca/__init__.py]: At this point, you might as well inline this method.

File: test/test.py [hunk 6]
```diff
@@ -158,6 +158,26 @@ def test_index_parameters(self):
...
+        args = {'file_or_dir': [file_path],
+                'replica': "aws",
+                'staging_bucket': "org-humancellatlas-dss-cli-test"}
+        response = api.upload(**args)
+
+        file_uuid = response['files'][0]['uuid']
+
+        process = subprocess.Popen(["hca", "get-files", file_uuid, "--replica", "aws"], stdout=subprocess.PIPE)
+        out, _ = process.communicate()
+        self.assertIsInstance(out, bytes)
```
Comment (ttung) [inline test/test.py]: Also, do you want to verify that the output is correct?

**Comment by ttung** [comment]
See comments pls.

File: hca/__init__.py (hunk 5 above)
Comment (Mackey22) [inline hca/__init__.py]: 👍

File: test/test.py (hunk 6 above)
Comment (Mackey22) [inline test/test.py]: 👍 https://github.com/HumanCellAtlas/data-store-cli/pull/32/commits/9f9dc2d2d33a0ccfc4d22ce348893725baf8973e
